.github/workflows/c-cpp.yml

```diff
@@ -10,19 +10,54 @@ on:
     branches: master
 
 jobs:
-  linux:
+  build:
     strategy:
       fail-fast: false
       matrix:
-        arch: [x64]
-        os: [ubuntu-20.04]
-        compiler:
-          - gcc
-          - clang
-    runs-on: ubuntu-latest
-    env:
-      CC: ${{ matrix.compiler }}
+        os:  [ubuntu-20.04, ubuntu-22.04]
+        cc: [gcc, clang]
+        include:
+          - os: macos-latest
+          - os: ubuntu-20.04
+            cc: slackware
+            container: 'andy5995/slackware-build-essential:15.0'
+
+    runs-on: ${{ matrix.os }}
+    container: ${{ matrix.container }}
     steps:
     - uses: actions/checkout@v3
-    - name: Build
+
+    - if: ${{ matrix.cc == 'clang' }}
+      run: |
+        echo "CC=clang" >> $GITHUB_ENV
+        echo "CXX=clang++" >> $GITHUB_ENV
+
+    - name: Show info
+      run: |
+        export -p
+        echo '${{ toJSON(matrix) }}'
+
+    # If the build tools ever seem to be out of date, this can be uncommented
+    #
+    #- if: ${{ contains(matrix.os, 'ubuntu') && matrix.container == null }}
+      #run: |
+        #sudo apt-get update -qq
+        #sudo apt-get upgrade -y -qq
+
+    - name: Make
       run: make
+
+  freebsd:
+    runs-on: macos-12
+    name: FreeBSD Test
+    steps:
+    - uses: actions/checkout@v3
+    - name: Test
+      uses: vmactions/freebsd-vm@v0
+      with:
+        usesh: true
+        prepare:  |
+          pkg install -y gmake
```

Comment:
@bbonev I don't understand all your notes. I'm installing gmake here, and running it below. Your docs indicate that gmake is required on *BSD

Reply:
Exactly, but for convenience you can run `make`. That will run `gmake` by itself...

I pushed some commits that fixed the builds on MacOS, FreeBSD, NetBSD and OpenBSD...

The one that fails because of `-flto=auto` can be fixed like this:

    make NO_FLTO=1 CFLAGS=-flto

Reply:
> Exactly, but for convenience you can run `make`. That will run `gmake` by itself...
> 
> I pushed some commits that fixed the builds on MacOS, FreeBSD, NetBSD and OpenBSD...
> 
> The one that fails because of `-flto=auto` can be fixed like this:
> 
> ```
> make NO_FLTO=1 CFLAGS=-flto
> ```

I added a check to the Makefile instead. Everything look ok?